services/src/handlers/wms.rs

```diff
@@ -214,9 +202,13 @@ where
         .body(response))
 }
 
-fn wms_url(external_address: &Url, workflow: WorkflowId) -> Result<Url> {
-    external_address
-        .join("wms/")?
+fn wms_url(workflow: WorkflowId) -> Result<Url> {
```

Comment:
I guess we had the external address as a parameter here to test this method. It once was buggy.

Reply:
the only thing that can be buggy using the wrong URL as a base and this wasn't actually covered by the tests at all. They merely checked whether the joining of the URL base with a path works and that's not even implemented by us but by the URL crate

Reply:
cf. above

Reply:
the logic of joining the url did not change, it's just that now the binding address is used if there is no external_address given. I could extract the method for joining the base with /wms/{id} but as I said that's trivial and already tested in the url crate

> Doesn't this remove everything from the root and then joins it?

no it does not. From the docs:

```   
  /// let base = Url::parse("https://example.net/a/b.html")?;
  /// let url = base.join("c.png")?;
  /// assert_eq!(url.as_str(), "https://example.net/a/c.png");
```